### ISSUE TITLE
Cannot add property "prpertyName", object is not extensible

### ISSUE THREAD

**Issue by abouroubi** [nartc/mapper#450]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

I updated my dependencies to the V8, and I'm having an issue, the mapped object created do not contain the properties that don't have the @AutoMap() decorator.

### Models/DTOs/VMs

This is my base model:
```typescript
export class BaseModel  {
  @AutoMap()
  id?: string;

  ttl?: number;
  deleted?: boolean;

  @AutoMap(() => Date)
  // When we create an object to add/update type us Date
  // When we get the object from the database CosmosDB will return a string
  createdAt?: Date | string;

  @AutoMap(() => Date)
  updatedAt?: Date | string;
}
```

This is my View Model
```typescript
export class BaseModelVm {
  @AutoMap()
  @IsDate()
  @IsOptional()
  createdAt?: Date;

  @AutoMap()
  @IsDate()
  @IsOptional()
  updatedAt?: Date;

  @AutoMap()
  @IsString()
  @IsOptional()
  id?: string;
}
```

### Mapping configuration

```typescript
  override get profile(): MappingProfile {
    return (mapper) => {
      createMap(
        mapper,
        BaseModel,
        BaseModelVm,
        forMember(
          (dest) => dest.createdAt,
          convertUsing(stringDateConverter, (src) => src.createdAt),
        ),
        forMember(
          (dest) => dest.updatedAt,
          convertUsing(stringDateConverter, (src) => src.updatedAt),
        ),
      );

      createMap(
        mapper,
        BaseModelVm,
        BaseModel,
        forMember(
          (dest) => dest.createdAt,
          convertUsing(stringDateConverter, (src) => src.createdAt),
        ),
        forMember(
          (dest) => dest.updatedAt,
          convertUsing(stringDateConverter, (src) => src.updatedAt),
        ),
      );
    };
  }
```

### Steps to reproduce

Call `const result = mapper.map(zone, BaseModelVm, BaseModel)`.

### Expected behavior

The result object is supposed to be an instance of BaseModel, so containing all the properties, but in this case the object will only contain `id`, `createdAt`, and `updatedAt`, and it omits the properties `deleted` and `ttl`.

And if I want to modify the `deleted` property by doing: 
```typescript 
result.deleted = false;
```

I have an exception:
```
Cannot add property deleted, object is not extensible
```  

I didn't not annotate deleted and id because they don't have to be mapped, but they need to exist in the object created.

Also why the objects instantiated by automapper are not extensible ? Is it voluntary ? 

### Package

- [ ] I don't know.
- [ ] <code>@automapper/core</code>
- [X] <code>@automapper/classes</code>
- [ ] <code>@automapper/nestjs</code>
- [ ] <code>@automapper/pojos</code>
- [ ] <code>@automapper/mikro</code>
- [ ] <code>@automapper/sequelize</code>
- [ ] Other (see below)

### AutoMapper version

8.1.0

**Comment by micalevisk** [comment]
did understand why you're using `stringDateConverter` if both destination and source has the type `Date` in both mapping (from `BaseModelVm` to `BaseModel`, and from `BaseModel` to `BaseModel`) :thinking:

**Comment by nartc** [comment]
@abouroubi question, in v7 would you still see `delete` and `ttl` get populated with `undefined`? I'd assume that they aren't set by AutoMapper in anyway because they're not decorated and they're not configured with `forMember`. From `AutoMapper` point of view, they're non-existent.

For `Cannot add property deleted, object is not extensible` exception, I understand the root cause.

AutoMapper has a default `postMap` that will be called with `Object.seal(destination)`, this prevents the consumers to add/delete properties to/from the `destination`. I can take off this constraint.

The workaround now is to use `forMember()` + `ignore()` for `delete` and `ttl`.

**Comment by a88zach** [comment]
Would it be possible to make this configurable?  Some may want want the mapped object to be sealed, but others may want to extend it after the mapping takes place.  An obvious use case is adding createdBy, createdAt fields to a mapped object before saving to a database.  This could be done with mapWithArguments, but that overly complicates the map

**Comment by abouroubi** [comment]
Thanks for the quick reply, I think I'll use the first work around.
Concerning the constraint that you want to remove, I wonder why you want to prevent the consumer from modifying the object ?

In v7.2.1 the object wasn't sealed so I had no issue when I used `result.deleted = false`

**Comment by nartc** [comment]
@abouroubi  @a88zach It's a bug I would say. The idea is in `postMap()` you shouldn't modify the `destination` because it might introduce unexpected behavior. I want to make sure the output of `map()` reflects the configuration and the metadata. However, after `mapper.map()` returns, you should be able to modify the result object. I'll fix it. Not sure about making it configurable but if needed, it is easy to add.

```ts
return postMap(Object.seal(destination)); // destination is already sealed. default postMap just returns the passed in seal object so the consumers get the sealed object 🤦‍♂️
```

**Comment by nartc** [comment]
@abouroubi  released 8.2.3 that should allows you to extend the object after map is finished
